tools/integration-tests/tester/tests/consensus/consensus_conflict_spam_test.go

```diff
@@ -0,0 +1,185 @@
+	require.NoError(t, err)
+	defer tests.ShutdownNetwork(ctx, t, n)
+
+	faucet, peer1 := n.Peers()[0], n.Peers()[1]
+	tokensPerRequest = faucet.Config().TokensPerRequest
+
+	tests.AwaitInitialFaucetOutputsPrepared(t, faucet, n.Peers())
+
+	fundingAddress := peer1.Address(0)
+	tests.SendFaucetRequest(t, peer1, fundingAddress)
+	require.Eventually(t, func() bool {
+		return tests.Balance(t, peer1, fundingAddress, ledgerstate.ColorIOTA) >= uint64(tokensPerRequest)
+	}, tests.Timeout, tests.Tick)
+
+	addresses := make([]*ledgerstate.Address, 100)
+	keyPairs := map[string]*ed25519.KeyPair{}
+	for i := 0; i < 100; i++ {
+		address := peer1.Address(i)
+		addresses[i] = &address
+		keyPairs[address.String()] = peer1.KeyPair(uint64(i))
+	}
+
+	outputs := getOutputsControlledBy(t, peer1, fundingAddress)
+	fundingKeyPair := map[string]*ed25519.KeyPair{fundingAddress.String(): peer1.KeyPair(0)}
+	outputs = splitToAddresses(t, peer1, outputs[0], fundingKeyPair, addresses...)
+	const conflictRepetitions = 4
+	pairwiseOutputs := outputs[:(conflictRepetitions-1)*3+3]
+	tripletOutputs := outputs[len(pairwiseOutputs):]
+
+	txs := []*ledgerstate.Transaction{}
+	for i := 0; i < conflictRepetitions; i++ {
+		sendPairWiseConflicts(t, n.Peers(), pairwiseOutputs[i*3:i*3+3], keyPairs, &txs, i)
+		sendTripleConflicts(t, n.Peers(), tripletOutputs[i*3:i*3+3], keyPairs, &txs, i)
+	}
+	t.Logf("number of txs to verify is %d", len(txs))
+	verifyConfirmationsOnPeers(t, n.Peers(), txs)
+}
+
+func verifyConfirmationsOnPeers(t *testing.T, peers []*framework.Node, txs []*ledgerstate.Transaction) {
+	const unknownGoF = 10
+	for _, tx := range txs {
+		// current value signifies that we don't know what is the previous gof
+		var prevGoF gof.GradeOfFinality = unknownGoF
+		for i, peer := range peers {
+			var metadata *jsonmodels.TransactionMetadata
+			var err error
+			require.Eventually(t, func() bool {
+				metadata, err = peer.GetTransactionMetadata(tx.ID().Base58())
+				return err == nil && metadata != nil
+			}, 10*time.Second, time.Second, "Peer %s can't fetch metadata of tx %s. metadata is %v. Error is %w",
+				peer.Name(), tx.ID().Base58(), metadata, err)
+			t.Logf("GoF is %d for tx %s in peer %s", metadata.GradeOfFinality, tx.ID().Base58(), peer.Name())
+			if prevGoF != unknownGoF {
+				require.EqualValues(t, prevGoF, metadata.GradeOfFinality,
+					"Different gofs on tx %s between peers %s and %s", tx.ID().Base58(),
+					peers[i-1].Name(), peer.Name())
+			}
+			prevGoF = metadata.GradeOfFinality
+		}
+	}
+}
+
+/**
+sendPairWiseConflicts receives a list of outputs controlled by a peer with certain peer index.
+It send them all to addresses controlled by the next peer, but it does so several time to create pairwise conflicts.
+The conflicts are TX_B<->TX_A<->TX_C
+*/
+func sendPairWiseConflicts(t *testing.T, peers []*framework.Node, outputs ledgerstate.Outputs,
+	keyPairs map[string]*ed25519.KeyPair,
+	txs *[]*ledgerstate.Transaction, iteration int) {
```

Comment:
A pointer to an array of pointers to Transactions? Arrays are pointers already.

Reply:
Do you need the concept of `iteration` here? Does this function really need to know about what step of the for loop you are calling it from?

Reply:
1. You meant slices are pointers. Since we are appending to the slice and passing it a long functions it must be a pointer to a slice. This is because a slice is not exactly a pointer. It is a pointer + header (that consists of length and capacity). Since we are appending to the slice we shouldn't be only chaging the different header copies.

2. Curerently `iteration` is good for logging and `determineTargets` function needs it

Reply:
1. Having this txs pointer so you can append doesn't seem very readable to me. Maybe you can append in the caller and have this function returned the txs it created?
2. Same principle here: you are dealing with iteration of a for loop in the caller inside the called function: I find it hard to read. Maybe the log messages can be factored out as part of the main loop?

Reply:
1. done
2. The problem is that `determineTargets` needs it to be passed down. Maybe I can give it a different name in the signature that is more meaningful like `startingPoint`. As for logs, I can take them out. But I will still have to pass the variable down, so not sure it will have the effect you wanted.